odbc/src/api/functions.rs

```diff
@@ -1655,8 +1723,35 @@ mod util {
     use odbc_sys::{Integer, SmallInt, SqlReturn, WChar};
     use std::{cmp::min, ptr::copy_nonoverlapping};
 
+    /// input_wtext_to_string converts an input cstring to a rust String.
+    /// It assumes nul termination if the supplied length is negative.
+    #[allow(clippy::uninit_vec)]
+    pub fn input_wtext_to_string(text: *const WChar, len: usize) -> String {
+        if (len as isize) < 0 {
+            let mut dst = Vec::new();
+            let mut itr = text;
+            unsafe {
+                while *itr != 0 {
+                    dst.push(*itr);
+                    itr = itr.offset(1);
+                }
+            }
```

Comment:
For my own understanding, why does this need to be in an `unsafe` block?

Reply:
You can only dereference raw pointers in unsafe blocks. In fact, this is generally the only place you need unsafe (there are a few others, but that's the most common). You can do whatever you want to a pointer whenever you want as long as it's never dereferenced (adding to a pointer is safe, so long as you do not dereference it).